### PR TITLE
Update WSV and Migration to Scale

### PR DESCRIPTION
**Pull request by humb1t**
### Description of the Change

Message passing model of WSV update and ISI processing.

### CODE REVIEW

File: iroha/src/account.rs [hunk 1]
```diff
@@ -293,25 +263,61 @@ pub mod isi {
...
+
+        #[test]
+        fn create_role_command_serialization_and_deserialization() {
+            let expected = CreateRole {
+                role_name: "user".to_string(),
+                permissions: Vec::new(),
+            };
+            let actual = <CreateRole>::decode(&mut expected.encode().as_slice()).unwrap();
+            assert_eq!(expected, actual);
+        }
```
Comment (kamilsa) [inline iroha/src/account.rs]: All these tests are doing almost the same. Aren't there a way in rust to write a templated test to implement test logic only once?

Comment (eadventurous) [reply iroha/src/account.rs]: Right now only with addition of external testing framework. Maybe [stainless](https://github.com/reem/stainless) framework is worth considering.

Comment (eadventurous) [reply iroha/src/account.rs]: But well, latest commit was 11 months ago... Though I haven't seen any other decent non standard testing framework.

Comment (eadventurous) [reply iroha/src/account.rs]: Also we can use [laze_static](https://docs.rs/lazy_static/1.4.0/lazy_static/) to mimic before all data initialization.

File: iroha/src/account.rs [hunk 2]
```diff
@@ -293,25 +263,61 @@ pub mod isi {
...
+    mod tests {
+        use super::*;
+
+        #[test]
+        fn append_role_command_serialization_and_deserialization() {
+            let expected = AppendRole {
+                account_id: Id::new("account", "domain"),
+                role_name: "role".to_string(),
+            };
+            let actual = <AppendRole>::decode(&mut expected.encode().as_slice()).unwrap();
```
Comment (eadventurous) [inline iroha/src/account.rs]: Does `AppendRole` need to be in `<..>`? I believe it would work without them, simply like `AppendRole::decode(...)`.

Comment (eadventurous) [reply iroha/src/account.rs]: Same for other test examples.

File: iroha/src/kura.rs [hunk 3]
```diff
@@ -1,73 +1,68 @@
...
+
+static DEFAULT_BLOCK_STORE_LOCATION: &str = "./blocks/";
 
 /// High level data storage representation.
 /// Provides all necessary methods to read and write data, hides implementation details.
 pub struct Kura {
+    mode: String,
     block_store: BlockStore,
-    world_state_view: WorldStateView,
+    world_state_view_tx: UnboundedSender<Block>,
```
Comment (eadventurous) [inline iroha/src/kura.rs]: Maybe introduce a type alias for `UnboundedSender<Block>`?

File: iroha/src/validation.rs [hunk 4]
```diff
@@ -13,9 +13,15 @@ pub struct MerkleTree {
...
+    pub fn new() -> Self {
+        MerkleTree {
+            root_node: Node::Empty,
+        }
+    }
+
     /// Builds a Merkle Tree from sorted array of `Blocks`.
     //TODO: should we check or sort blocks here?
-    pub fn build(blocks: &[&Block]) -> Self {
+    pub fn build(&mut self, blocks: &[&Block]) {
```
Comment (eadventurous) [inline iroha/src/validation.rs]: Haven't noticed before, but I think we should change `&[&Block]` type. Why use slice of references if it's itself a reference? I think it unnecessarily complicates this type.

File: iroha/src/account.rs (hunk 1 above)
Comment (humb1t) [reply iroha/src/account.rs]: me and @eadventurous think about code generation for ISIs - https://soramitsu.atlassian.net/browse/RH2-6

File: iroha/src/account.rs (hunk 2 above)
Comment (humb1t) [reply iroha/src/account.rs]: Both options work, IDK why they use `<..>` notation in docs

Comment (eadventurous) [reply iroha/src/account.rs]: I believe they use them in docs when specifying generic types, at least that's what I got from there examples.

File: iroha/src/validation.rs (hunk 4 above)
Comment (humb1t) [reply iroha/src/validation.rs]: @eadventurous I assume that right now we don't have guarantees about mutability of blocks passed between different modules and using slice of references I try to protect API from algorithmic [connascence](https://en.wikipedia.org/wiki/Connascence).
(More about the term: https://www.youtube.com/watch?v=fSr8LDcb0Y0)

**Pull request merged**